src/foam/blob/BlobJava.js

```diff
@@ -210,3 +210,155 @@ return getParent().read(buffer, offset + getOffset());`
     }
   ]
 });
+
+foam.CLASS({
+  refines: 'foam.blob.BlobStore',
+
+  javaImports: [
+    'foam.core.X',
+    'org.apache.geronimo.mail.util.Hex',
+    'java.io.File',
+    'java.io.FileOutputStream',
+    'java.nio.ByteBuffer',
+    'java.security.MessageDigest'
+  ],
+
+  constants: [
+    {
+      name: 'BUFFER_SIZE',
+      value: 8192,
+      type: 'int'
+    }
+  ],
+
+  properties: [
+    { class: 'String', name: 'tmp', javaFactory: 'return getRoot() + File.separator + "tmp";' },
+    { class: 'String', name: 'sha256', javaFactory: 'return getRoot() + File.separator + "sha256";' }
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode:
+`if ( blob instanceof IdentifiedBlob ) {
+  return blob;
+}
+
+this.setup();
+
+try {
+  MessageDigest hash = MessageDigest.getInstance("SHA-256");
+  Buffer buffer = new Buffer(BUFFER_SIZE, ByteBuffer.allocate(BUFFER_SIZE));
+
+  long chunk = 0;
+  long size = blob.getSize();
+  long chunks = (long) Math.ceil((double) size / (double) BUFFER_SIZE);
+
+  File tmp = allocateTmp(1);
+  while (chunk < chunks) {
```

Comment:
foam coding standard is:
while ( chunk < chunks ) { (extra spaces)

Reply:
Thanks. Modified